Feat/garage sign auto version

patrickvacek opened the pull request:
Use a script to fetch garage-sign instead of hardcoding a specific version.

Define `GARAGE_SIGN_VERSION` to control the version to download and package, e.g. `cmake ../ -DGARAGE_SIGN_VERSION=cli-0.3.0-5-g5908997.tgz`

Define `GARAGE_SIGN_SHA256` to provide a hash to check the downloaded file, e.g. `cmake ../ -DGARAGE_SIGN_SHA256=a24efff06bbe1617947d4a08a74f617af9a306d7013f0881407df9d630253b2b`
    
The default is the version whose name sorts highest, which should be the greatest version number. If the layout or content of the tuf-cli releases page changes substantially, this may break.

lbonn commented on scripts/get_garage_sign.py:
```diff
@@ -0,0 +1,110 @@
+    # Assume namespace.
+    ns = '{http://s3.amazonaws.com/doc/2006-03-01/}'
+    items = tree.findall(ns + 'Contents')
+    versions = dict()
```
nitpick but perhaps too many list comprehension:

```
for i in items:
    if not i.find(ns + 'Key').text.startswith('cli-'):
        continue
    ...
```

Or use an intermerdiary variable

```
cli_items = [i for i in items if i.find(ns + 'Key').startswith('cli-')]
for i in cli_items:
    ...
```

lbonn replied:
They should be equivalent functionally except if I made a fatal mistake, this is only about readability, which is often subjective. I tested the first version by running the program without argument and it successfully downloaded the archive.

Other than that, it's only a matter of style, really... The first version looks more like traditional imperative programming.

patrickvacek commented:
Rebased to fix merge conflicts from the recent C++11 PR.

cajun-rat commented:
Agreed, lets go with this.  I'm having oe-selftest failures on shovel at the moment, not sure if they are due to this or not...

patrickvacek commented:
> I'm having oe-selftest failures on shovel at the moment, not sure if they are due to this or not...

Are you using https://github.com/advancedtelematic/meta-updater/pull/291? If so, I just realized it was slightly stale, so I've fixed that now. If not, you probably need to, since I changed the location of garage-sign. If that doesn't fix the problem, let me know.

And for the record, I'm not opposed to looking at other/better ways of doing this as well. But for now I think this should be an improvement.

cajun-rat commented:
I'd missed the meta-updater change, that fixes it!
